### ISSUE TITLE
RFC: Rewind feature for Games using OrbitDB

### ISSUE THREAD

**Issue by vasa-develop** [simpleaswater/rfcs#16]
**What new topic/thread/tutorial are you proposing? Please describe.**
Proposed By @aphelionz: I was playing around using OrbitDB for games in my spare time using an entity component system architecture and one thing that orbit could probably do really well is a "rewind" feature like in Braid.

**Comment by haadcode** [comment]
Hey @vasa-develop and @aphelionz  would love to understand the use case better. OrbitDB, via ipfs-log, supports "rewind" and in fact imo is one of the best features of immutable logs, albeit underutilized and not necessarily intuitively exposed to the user in OrbitDB. Would love to understand what you'd need for "rewind", we can probably make it happen :smile: :v: 

Huge fan of EC systems personally and rooting to see OrbitDB used the in one :+1: Happy to help if anyone's working on it.

**Comment by aphelionz** [comment]
Yeah! I had envisioned this basically by using either the log or the docstore abstraction on top of it, with some performance improvements.

You could do it with the log by simply saving the game state as a payload for each entry, OR

You could do it with the docstore with each game entity getting its own document and then performing the rewind on top of that.
